src/pages/audio-rewards-page/components/modals/CognitoModal.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useEffect } from 'react'
+
+import { useDispatch } from 'react-redux'
+
+import { useModalState } from 'common/hooks/useModalState'
+import {
+  FlowUIOpenEvent,
+  FlowUICloseEvent,
+  FlowSessionEvent,
+  FlowErrorEvent
+} from 'common/models/AudioRewards'
+import { getUserHandle } from 'common/store/account/selectors'
+import {
+  CognitoFlowStatus,
+  setCognitoFlowStatus
+} from 'common/store/pages/audio-rewards/slice'
+import { useScript } from 'hooks/useScript'
+import AudiusBackend from 'services/AudiusBackend'
+import { COGNITO_SCRIPT_URL } from 'utils/constants'
+import { useSelector } from 'utils/reducer'
+
+declare global {
+  interface Window {
+    Flow: any
+  }
+}
+
+/**
+ * The CognitoModal isn't a true modal. It uses the Cognito Flow SDK to trigger an externally controlled modal
+ */
+export const CognitoModal = () => {
+  const dispatch = useDispatch()
+  const [isOpen] = useModalState('Cognito')
+  const scriptLoaded = useScript(COGNITO_SCRIPT_URL)
+  const handle = useSelector(getUserHandle)
+
+  const triggerFlow = useCallback(async () => {
+    const { signature } = await AudiusBackend.getCognitoSignature()
+    if (!signature) {
+      console.error('COGNITO: Failed to get Cognito signature')
+      return
+    }
+    const flow = new window.Flow({
+      publishableKey: process.env.REACT_APP_COGNITO_KEY,
+      templateId: process.env.REACT_APP_COGNITO_TEMPLATE_ID,
+      user: {
+        customerReference: handle,
+        signature
+      }
+    })
+
+    flow.on('ui', (event: FlowUIOpenEvent | FlowUICloseEvent) => {
+      switch (event.action) {
+        case 'opened':
+          dispatch(setCognitoFlowStatus({ status: CognitoFlowStatus.OPENED }))
+          break
+        case 'closed':
+          dispatch(setCognitoFlowStatus({ status: CognitoFlowStatus.CLOSED }))
+          break
+        default:
```

Comment:
wondering if we need a default here, i think if ts sees that 'opened' and 'closed' are the only types for `event.action`, then default is not needed?